kats/src/main/kotlin/kats/Reader.kt

```diff
@@ -0,0 +1,16 @@
+package kats
+
+/**
+ * Basic implementation of the Reader monad. Provides an "implicit" context (configuration) for
+ * function execution. Intended to provide Dependency Injection.
+ */
+class Reader<C, A>(val rd: (C) -> A) {
```

Comment:
`A` should be covariant and the `C` contravariant and both types should be hierarchies from `Any`